osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/misc/GetGsmDiagnosticCommandExecutorTest.java

```diff
@@ -0,0 +1,263 @@
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ModemRegistrationStatusDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.PacketSwitchedStatusDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.SignalQualityDto;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class GetGsmDiagnosticCommandExecutorTest {
+
+  @InjectMocks private GetGsmDiagnosticCommandExecutor executor;
+
+  @Mock private DlmsMessageListener dlmsMessageListener;
+
+  @Mock private DlmsHelper dlmsHelper;
+
+  @Mock private DlmsObjectConfigService dlmsObjectConfigService;
+
+  @Mock private DlmsConnectionManager connectionManager;
+
+  private final DlmsDevice device = this.createDevice(Protocol.SMR_5_1, CommunicationMethod.CDMA);
+  private final long from = 1111111L;
+  private final long to = 2222222L;
+  private final int classId = 99;
+  private final String obiscode = "1.2.3.4.5.6";
+  private final GetGsmDiagnosticRequestDto request = new GetGsmDiagnosticRequestDto();
+
+  @BeforeEach
+  public void setUp() {
+    when(this.connectionManager.getDlmsMessageListener()).thenReturn(this.dlmsMessageListener);
+  }
+
+  @Test
+  public void testExecuteObjectNotFound() throws ProtocolAdapterException {
+    // SETUP
+    when(this.dlmsObjectConfigService.findDlmsObjectForCommunicationMethod(any(), any()))
+        .thenThrow(new ProtocolAdapterException("Object not found"));
+
+    // CALL
+    try {
+      this.executor.execute(this.connectionManager, this.device, this.request);
+      fail("When no matching object is found, then execute should fail");
+    } catch (final ProtocolAdapterException e) {
+      assertThat(e.getMessage()).isEqualTo("Object not found");
+    }
+  }
+
+  @Test
+  public void testBundle() throws ProtocolAdapterException {
+    // CALL
+    final GetGsmDiagnosticRequestDto request = this.executor.fromBundleRequestInput(this.request);
+
+    // VERIFY
+    // Verify that the executor creates a new request
+    assertThat(request).isNotEqualTo(this.request);
+  }
+
+  @Test
+  public void testHappy() throws Exception {
+
+    // SETUP
+    when(this.dlmsObjectConfigService.findDlmsObjectForCommunicationMethod(
+            this.device, DlmsObjectType.GSM_DIAGNOSTIC))
+        .thenReturn(new DlmsObject(DlmsObjectType.GSM_DIAGNOSTIC, this.classId, this.obiscode));
+
+    // SETUP - mock dlms helper to return data objects on request
+    final AttributeAddress[] addresses = {
+      new AttributeAddress(this.classId, this.obiscode, 2),
+      new AttributeAddress(this.classId, this.obiscode, 3),
+      new AttributeAddress(this.classId, this.obiscode, 4),
+      new AttributeAddress(this.classId, this.obiscode, 5),
+      new AttributeAddress(this.classId, this.obiscode, 6),
+      new AttributeAddress(this.classId, this.obiscode, 7)
+      // new AttributeAddress(this.classId, this.obiscode, 8)
+    };
```

Comment:
These aren't used, and should be removed or used as a result of some other modification.

Reply:
Removed